modules/renter/downloadcache_test.go

```diff
@@ -1,36 +1,83 @@
 package renter
 
 import (
+	"container/heap"
 	"strconv"
-	"sync"
 	"testing"
 	"time"
 )
 
-// TestAddChunkToCache tests that the oldest chunk is removed
-func TestAddChunkToCache(t *testing.T) {
+// TestAdd tests that the oldest chunk is removed
+func TestAdd(t *testing.T) {
+	// Initializing minimum variables
 	udc := &unfinishedDownloadChunk{
 		download: &download{
 			staticDestinationType: destinationTypeSeekStream,
 		},
-		chunkCache: make(map[string]*cacheData),
-		cacheMu:    new(sync.Mutex),
+		downloadChunkCache: new(downloadChunkCache),
 	}
 
+	udc.downloadChunkCache.init()
+
+	// Testing Push to Heap
```

Comment:
I'd rename `TestAdd`to `TestStreamCache` and test the cache using its exported high-level methods which add chunks, change the cacheSize etc. It should also check if the least-recently-accessed element is removed and not merely the least-recently-added.
Then you could create a second low-level test `TestHeapImplementation` that checks if your custom heap works as expected (`Push`, `Pop`, `update`).